### PR TITLE
Add K2 Maze scenario

### PR DESCRIPTION
**Pull request by RedRafe**
## Changes:
- [x] Added Krastorio2 maze scenario (based off maze scenario already present)
  - tweaks Market's items (if market is present) to add/remove things from vanilla to K2
  - uses Dedalus's mythology as theme for the scenario/description
- [x] Removed the flag `button.enabled = false` for the biter evo button in top gui because even if the button does nothing, setting it to disabled makes it look different from all other buttons
- [x] added a function to switch the score sprite based on mods present: K2 removes that achievement so any scenario with K2 and score gui would crash on load because the sprite is not available anymore.

Here's a preview:
Screenshot from 2024-07-31 10-56-27

Merge is not urgent but this PR is good to go on my part: I've already playtested the scenario multiple times

### CODE REVIEW

File: features/gui/score.lua [hunk 1]
```diff
@@ -75,6 +75,13 @@ local function schedule_redraw_score()
     memory.redraw_score_scheduled = true
 end
 
+local function get_modded_sprite()
```
Comment (grilledham) [inline features/gui/score.lua]:
```suggestion
local function get_score_sprite()
```

**Review by grilledham — APPROVED** [review]
LGTM. Do you want to add it to scenario templates?

**Comment by RedRafe** [comment]
> Do you want to add it to scenario templates?

Mmh, IDK. I don't think so (mainly because `maze.lua` does not have its own template either) but could always add if you prefer, maybe it's useful when launching from interface if we want to play it soon? I'm good either ways

**Comment by grilledham** [comment]
> > Do you want to add it to scenario templates?
> 
> Mmh, IDK. I don't think so (mainly because `maze.lua` does not have its own template either) but could always add if you prefer, maybe it's useful when launching from interface if we want to play it soon? I'm good either ways

The original maze map predates the scenario templates folder and I didn't retroactively add the old maps. There is no cost to adding maps to scenario templates, so I think it makes sense for the maps we are planning to play. Even if we think it's only going to be once, another admin might want to launch the map at a later date and it's easier if they can do that from the web interface rather than having to make a save and upload it.

**Comment by RedRafe** [comment]
Well lets add all mazes now so that's done

**Pull request merged**